src/src/com/microsoft/aad/adal/StorageHelper.java

```diff
@@ -264,28 +240,27 @@ public String decrypt(String value) throws GeneralSecurityException, IOException
 
         // get key version used for this data. If user upgraded to different
         // API level, data needs to be updated
-        String keyVersionCheck = new String(bytes, 0, KEY_VERSION_BLOB_LENGTH,
+        String encryptedDataKeyVersion = new String(bytes, 0, KEY_VERSION_BLOB_LENGTH,
                 AuthenticationConstants.ENCODING_UTF8);
-        Logger.v(TAG, "Encrypt version:" + keyVersionCheck);
+        Logger.v(TAG, "Encrypt version:" + encryptedDataKeyVersion);
 
-        SecretKey versionKey = getKeyForVersion(keyVersionCheck);
-        SecretKey versionMacKey = getMacKey(versionKey);
+        SecretKey secretKey = loadSecretKeyForDecryption(encryptedDataKeyVersion);
+        SecretKey macKey = getMacKey(secretKey);
 
         // byte input array: encryptedData-iv-macDigest
         int ivIndex = bytes.length - DATA_KEY_LENGTH - MAC_LENGTH;
         int macIndex = bytes.length - MAC_LENGTH;
         int encryptedLength = ivIndex - KEY_VERSION_BLOB_LENGTH;
         if (ivIndex < 0 || macIndex < 0 || encryptedLength < 0) {
-            throw new IllegalArgumentException(
-                    "Given value is smaller than the IV vector and MAC length");
+            throw new IllegalArgumentException("Invalid byte array input for decryption.");
```

Comment:
If we're throwing IOExceptiion, it will be bubbled up to the decrypt method where we've added the telemetry events, no need to add it here.